### PR TITLE
Migrate from circle ci to github ci

### PR DESCRIPTION
**Pull request by manoranjith**

### CODE REVIEW

File: .github/workflows/build_deploy.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_dir: ./public
+        force_orphan: true
+        user_name: "github-ci"
+        user_email: "github-ci@nomail.com"
+	# && git commit $(_COMMITFLAG) \
+	#               -m "HTML build of $(SPHINXPROJ) with 'make deploy'" \
+	#               -m "Source: $(_REMOTEURL)/tree/$(_GIT_COMMIT)" \
```
Comment (matthiasgeihs) [inline .github/workflows/build_deploy.yml:39]: Remove these lines?

File: Makefile [hunk 2]
```diff
@@ -89,6 +88,7 @@ check:
 	fi
 
 deploy: check stage ## to deploy staged artefacts to github pages repository
+	cp -a .circleci "$(STAGEDIR)"
```
Comment (matthiasgeihs) [inline Makefile:91]: Can be removed?

Comment (manoranjith) [reply Makefile]: I have removed this in this [force push](https://github.com/hyperledger-labs/perun-doc/compare/60378e6341772a29e06cbeea7aed89ccef78a852..f3da248e6aee21b5825a847b059912427d500b45).

File: .github/workflows/build_deploy.yml (hunk 1 above)
Comment (manoranjith) [reply .github/workflows/build_deploy.yml]: I have replaced this proper options for adding a commit message in this [force push](https://github.com/hyperledger-labs/perun-doc/compare/60378e6341772a29e06cbeea7aed89ccef78a852..f3da248e6aee21b5825a847b059912427d500b45).

**Review by matthiasgeihs — APPROVED** [review]

**Pull request merged**